clang/test/Driver/clang-offload-wrapper-multi-batch.c

```diff
@@ -0,0 +1,274 @@
+// CHECK-DISASM:      __sycl_offload_entry_name{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: _entry_targ_1_BB
+
+// CHECK-DISASM:      @.sycl_offloading.target.{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: target-one
+// CHECK-DISASM:      @prop{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: testProp-name13
+// CHECK-DISASM:      @SYCL_PropSetName{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: testProp-GroupF
+// CHECK-DISASM:      @.sycl_offloading.{{.*}}.data = internal unnamed_addr constant
+// CHECK-DISASM-SAME: __CLANG_OFFLOAD_BUNDLE__sycl-target-one
+// CHECK-DISASM:      __sycl_offload_entry_name{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: _entry_targ_1_CC
+
+// CHECK-DISASM:      @.sycl_offloading.target.{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: target-two
+// CHECK-DISASM:      @prop{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: testProp-name17
+// CHECK-DISASM:      @SYCL_PropSetName{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: testProp-GroupG
+// CHECK-DISASM:      @.sycl_offloading.{{.*}}.data = internal unnamed_addr constant
+// CHECK-DISASM-SAME: __CLANG_OFFLOAD_BUNDLE__sycl-target-two
+// CHECK-DISASM:      __sycl_offload_entry_name{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: _entry_targ_2_AAA
+
+// CHECK-DISASM:      @.sycl_offloading.target.{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: target-two
+// CHECK-DISASM:      @prop{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: testProp-name19
+// CHECK-DISASM:      @SYCL_PropSetName{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: testProp-GroupH
+// CHECK-DISASM:      @.sycl_offloading.{{.*}}.data = internal unnamed_addr constant
+// CHECK-DISASM-SAME: __CLANG_OFFLOAD_BUNDLE__sycl-target-two
+// CHECK-DISASM:      __sycl_offload_entry_name{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: _entry_targ_2_BBB
+
+// CHECK-DISASM:      @.sycl_offloading.target.{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: target-two
+// CHECK-DISASM:      @prop{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: testProp-name23
+// CHECK-DISASM:      @SYCL_PropSetName{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: testProp-GroupI
+// CHECK-DISASM:      @.sycl_offloading.{{.*}}.data = internal unnamed_addr constant
+// CHECK-DISASM-SAME: __CLANG_OFFLOAD_BUNDLE__sycl-target-two
+// CHECK-DISASM:      __sycl_offload_entry_name{{.*}} = internal unnamed_addr constant
+// CHECK-DISASM-SAME: _entry_targ_2_CCC
+
+///////////////////////////////////////////////////////////////////////////////////////////
+// Generate the same wrapped BC file with a single "-batch" option.  This shows that
+// "-batch" is a mode setting for all input, not an option to introduce a batch/table file.
+// Note that "-batch" even affects prior input (e.g. %t.0.table).
```

Comment:
Is there any issue with one target using `-batch` which would expect the table input, but other targets that do not?  I'm not suggesting that there is a scenario like this (where one target does splitting and another doesn't) but I'm curious as to what would happen if that case does exist.

Reply:
That case would not work.  -batch causes all input to be treated as a table file.    If mixed mode (i.e. table and non-table) input is necessary, clang-offload-wrapper will need to be updated.

Reply:
If you could add a comment in the wrapper code about the -batch usage requirements that would be great.  Or maybe info in the -help section to state -batch applies to all targets.

Reply:
Comment and help message updated.  Help message is now:

```
  --batch                        - All input files are treated as a table file.  One table file per target.
                                   Table files consist of a table of filenames that provide
                                   Code, Symbols, Properties, etc.
                                   Example input table file in batch mode:
                                     [Code|Symbols|Properties|Manifest]
                                     a_0.bc|a_0.sym|a_0.props|a_0.mnf
                                     a_1.bin|||
                                   Example usage:
                                     clang-offload-wrapper -batch -host=x86_64-unknown-linux-gnu
                                       -kind=openmp -target=spir64_gen table1.txt
                                       -kind=openmp -target=spir64     table2.txt
```